app/controllers/sitemaps_controller.rb

```diff
@@ -0,0 +1,22 @@
+class SitemapsController < ApplicationController
+  before_action :set_cache_control_headers, only: %i[show]
+
+  def show
+    date_string = params[:sitemap].gsub("sitemap-", "").gsub(".xml", "")
+    not_found unless date_string.match?(/\A[A-Z][a-z][a-z]\-\d{4}\z/)
+    begin
+      date = Time.zone.parse(date_string).at_beginning_of_month
+    rescue ArgumentError
+      not_found
+    end
+    end_date = (date + 1.month).at_beginning_of_month
```

Comment:
curious to know why we're specifically using +1 month `at_beginning_of_month` rather than `end_of_month`? is there a specific edge case we need to consider here?

Reply:
Me forgetting that was a thing. I think I'm thrown off by `end_of_month` and `at_beginning_of_month` having different formats. Why not `at_...` for both, or neither? 🤷‍♂